prow/caching.md

```diff
@@ -0,0 +1,50 @@
+# Caching dependencies in Prow
```

Comment:
I am curious about the choice to use GCS as the cache store rather than local disk. There are definitely some trade offs, seems disk is faster (not sure how much), simpler, but has potential size constraints (for Kubernetes scale - no where near a problem for Istio yet), and higher cache miss (new node is empty).

Reply:
@howardjohn that assumes thread safety of the cache, is this the case for golang? I think it would be a bit safer to generally not assume that

Reply:
I am fairly sure it is for go, but may not be for other cache usages

Reply:
I've opened https://github.com/golang/go/issues/40461 to find out if golangs build and test cache is threadsafe

Reply:
To what are we referring when speaking about "local disk"? A `PersistentVolume`? I think this would work as well, but the benefit of using a GCS bucket I saw were:
- we have the chance to publicly download the cache if we want to debug the build
- we can enable some sort of lifecycle to cleanup old caches automatically